### PR TITLE
Added support for individualized radius values in :meth:`.Polygram.round_corners` 

### PR DESCRIPTION
**Pull request by coreyp1**
rounded_ManimCE_v0 17 2

## Overview: What does this pull request change?
The pull request extends the `Polygram::round_corners()` function to accept additional/enhanced arguments.

The new features are as follows:

* If the `radius` arguent is supplied as a single value (original behavior), then the same radius will be applied to all corners.  If `radius` is a list, then the individual values will be applied sequentially, with the first corner receiving `radius[0]`, the second corner receiving `radius[1]`, etc.  The radius list will be repeated as necessary.

* The `components_per_rounded_corner` argument (new) is provided so that the fidelity of the rounded corner may be fine-tuned as needed.  2 is an appropriate value for most shapes, however a larger value may be need if the rounded corner is particularly large.  2 is the minimum number allowed, representing the start and end of the curve.  3 will result in a start, middle, and end point, meaning 2 curves will be generated.

* The `evenly_distribute_anchors` argument (new) is provided so that the line segments (the part part of each line remaining after rounding off the corners) can be subdivided to a density similar to that of the average density of the rounded corners.  This may be desirable in situations in which an even distribution of curves is desired for use in later transformation animations.  Be aware, though, that enabling this option can result in an an object containing significantly more points than the original, especially when the rounded corner curves are small.

The `RoundedRectangle` test was updated since there were sub-pixel differences in the original default behavior (8 bezier curves per rounded corner) vs. the new default behavior (1 bezier curve per rounded corner).

## Motivation and Explanation: Why and how do your changes improve the library?
* Before, all corner radiuses had to be the same.  This expands on that ability.
* Before, having a lot of rounded edge objects was computationally taxing to generate due to the excessive number of vertices that were generated by (1) the rounded corner itself, and (2) the attempt to add extra vertices to each straight line connecting the rounded corners.  This was especially eggregious when using a small rounded corner relative to the shape.  If too small, it generates Out Of Memory errors (trying to allocate petabytes... as described here #3052 .

## Links to added or changed documentation pages
Changed:
* https://manimce--3155.org.readthedocs.build/en/3155/reference/manim.mobject.geometry.polygram.Polygram.html?highlight=round_corners#manim.mobject.geometry.polygram.Polygram.round_corners

## Further Information and Comments

* A previous issue (https://github.com/ManimCommunity/manim/issues/3052) reported memory issues caused by attempts to evenly distribute the anchors.  This puts that behavior behind a flag which disables the even distribution by default.  It also protects against a similar OOM issue when the radius is 0 for a specific corner.  Setting the behavior to disabled by default has 2 advantages: It greatly reduces the vertex count and it matches the behavior seen in the 3b1b repo.
* Previously, all rounded corner arcs were created using 8 bezier curves.  This reduces that to only 1 bezier curve by default (which matches the behavior in the 3b1b repo), but allows for more bezier curves if they are actually needed.
* The combination of the previous 2 changes reduced the vertex count of a default rounded rectangle from 100+ down to 8.  The performance is very much improved in generating these shapes as well.
* Lastly, in case you are curious, I often record my coding sessions when working on an open source project.  The video that goes along with this pull request is here: https://youtu.be/V2M2Jn3q5tc
* I'm new to Manim... please be kind. :)
* Thanks for the work you all put in on maintaining a great project!

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/mobject/geometry/polygram.py [hunk 1]
```diff
@@ -169,18 +204,28 @@ def construct(self):
...
         for vertices in self.get_vertex_groups():
             arcs = []
-            for v1, v2, v3 in adjacent_n_tuples(vertices, 3):
+
+            # Repeat the radius list as necessary in order to provide a radius
+            # for each vertex.
+            if type(radius) is float:
+                radius_list = [radius] * len(vertices)
+            else:
+                radius_list = radius * ceil(len(vertices) / len(radius))
```
Comment (behackl) [inline manim/mobject/geometry/polygram.py:213]: What happens if I pass `radius=1`? Given that `type(1) is float` yields `False`, you rather might want to use `isinstance(radius, (int, float))` instead.

File: manim/mobject/geometry/polygram.py [hunk 2]
```diff
@@ -133,18 +134,52 @@ def get_vertex_groups(self) -> np.ndarray:
...
         """Rounds off the corners of the :class:`Polygram`.
 
         Parameters
         ----------
-        radius
+        radius : float | list[float]
             The curvature of the corners of the :class:`Polygram`.
+        evenly_distribute_anchors : bool
+            Break long line segments into proportionally-sized segments.
+        components_per_rounded_corner : int
```
Comment (behackl) [inline manim/mobject/geometry/polygram.py:151]: Type hints added to the methods don't need to be repeated in the docstring.

```suggestion
        radius
            The curvature of the corners of the :class:`Polygram`.
        evenly_distribute_anchors
            Break long line segments into proportionally-sized segments.
        components_per_rounded_corner
```

**Review by behackl — CHANGES_REQUESTED** [review]
Hey, thanks for your contribution! This looks pretty neat, kudos on the nice implementation!

I believe to have found one issue with it; I've left a comment there -- and I've also left another suggestion regarding the docstring. Please take a look and let me know what you think!

File: manim/mobject/geometry/polygram.py (hunk 1 above)
Comment (coreyp1) [reply manim/mobject/geometry/polygram.py]: Applied!

File: manim/mobject/geometry/polygram.py (hunk 2 above)
Comment (coreyp1) [reply manim/mobject/geometry/polygram.py]: Applied!

**Review by behackl — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**